payroll/payment_gateway/payment_gateway_config.py

```diff
@@ -1,18 +1,39 @@
 import importlib
 
+from django.conf import settings
 from payroll.apps import PayrollConfig
 
 
 class PaymentGatewayConfig:
-    def __init__(self):
-        self.gateway_base_url = PayrollConfig.gateway_base_url
-        self.endpoint_payment = PayrollConfig.endpoint_payment
-        self.endpoint_reconciliation = PayrollConfig.endpoint_reconciliation
-        self.api_key = PayrollConfig.payment_gateway_api_key
-        self.basic_auth_username = PayrollConfig.payment_gateway_basic_auth_username
-        self.basic_auth_password = PayrollConfig.payment_gateway_basic_auth_password
-        self.timeout = PayrollConfig.payment_gateway_timeout
-        self.auth_type = PayrollConfig.payment_gateway_auth_type
+    """
+    Configuration handler for payment gateway integrations.
+    Supports payment point specific configurations with fallback to global settings.
+    """
+    def __init__(self, payment_point=None):
```

Comment:
@anthbel anthbel Since only payment_point.name is used consider passing in `payment_point-_name` as the argument instead of the payment_point dict.